update_bundle.py

```diff
@@ -17,10 +17,17 @@ def fetch_revision(charm, charm_channel):
     charm_info = requests.get(
         f"https://api.snapcraft.io/v2/charms/info/{charm}?fields=channel-map"
     ).json()
+    revision = 0
     for channel in charm_info["channel-map"]:
-        if channel["channel"]["risk"] == charm_channel:
-            return channel["revision"]["revision"]
-    raise ValueError("Revision not found.")
+        if (
+            channel["channel"]["risk"] == charm_channel
+            and channel["revision"]["revision"] > revision
+        ):
+            revision = channel["revision"]["revision"]
+    if revision:
+        return revision
+    else:
+        raise ValueError("Revision not found.")
```

Comment:
It's possible (and don't quote me on this) that you no longer need to specify bundle revisions. Try manually deploying them locally first and see.

My kafka bundles are failing due to permissions issues, but they look like the following: 
`publish` - https://github.com/canonical/kafka-bundle/blob/main/.github/workflows/publish.yaml
`bundle.yaml` - https://github.com/canonical/kafka-bundle/blob/main/releases/latest/k8s/kafka-k8s/bundle.yaml

Reply:
As I haven't been able to successfully publish *yet* you might see that this doesn't work, but can check that locally first.

Reply:
Let me check that! Thanks for the suggestion, Marc.

Reply:
I changed my mind. You don't need `bundle` revisions, but the charms in the bundle could still have them. Ignore these comments.

Reply:
I checked and it deploy the latest version (the right one). But, yeah, as you commented we can keep the exact version of each charm in the bundle file. I'll keep them for now.

Reply:
Bundle revisions are irrelevant to bundle deployments, because charmhub only recognises the _channel_ from which we deploy a bundle. However, Mykola asked me to keep bundle revisions around so we have regular testing, making it easier to pinpoint when issues are arising.

Reply:
Thanks for the context, Will!